builtin/logical/mysql/path_config_connection.go

```diff
@@ -32,8 +36,14 @@ func (b *backend) pathConnectionWrite(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
 	connString := data.Get("value").(string)
 
+	maxOpenConns := data.Get("max_open_connections").(int)
+	if maxOpenConns == 0 {
+		maxOpenConns = 2
+	}
```

Comment:
I know that not all backends do this, but since this one is pretty simple...

Check for the existence of a previous entry...if it exists, and if there isn't an actual value set here (use GetOk), don't overwrite it.

Do the same for the connection URL, so that they can adjust the max open connections without having to respecify that.